airflow/ui/src/components/FlexibleForm/FieldStringArray.tsx

```diff
@@ -19,15 +19,32 @@
 import { Textarea } from "@chakra-ui/react";
 
 import type { FlexibleFormElementProps } from ".";
+import { paramPlaceholder, useParamStore } from "../TriggerDag/useParamStore";
 
-export const FieldStringArray = ({ name, param }: FlexibleFormElementProps) => (
-  <Textarea
-    defaultValue={
-      Array.isArray(param.value) ? (param.value as Array<string>).join("\n") : String(param.value ?? "")
+export const FieldStringArray = ({ name }: FlexibleFormElementProps) => {
+  const { paramsDict, setParamsDict } = useParamStore();
+  const param = paramsDict[name] ?? paramPlaceholder;
+  const handleChange = (newValue: string) => {
+    const newValueArray = newValue.split("\n");
+
+    if (paramsDict[name]) {
+      paramsDict[name].value = newValueArray;
```

Comment:
Same like above. In old UI an empty array was returned as `null` here.

Reply:
...else the required field check will not fail in Schema checks actually.

Reply:
Just checked and the suspected state here is still "wrong". If you remove all text then a list with one element as empty string is used. Should e be `null` here as well to be like the legacy UI:

![image](https://github.com/user-attachments/assets/64687ad1-eb4f-4d52-a06c-b5804262e095)